average.js

```diff
@@ -1,6 +1,15 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  let ans = 0;//initialize answer
+  let len = numbers.length;//length of array
+  numbers.map((num,i)=>{
+    if(!isNaN(num)){
+      ans += num; //add value to ans if it is a number
+    }
+    else{
+      len -= 1 //if it is not a number, reduce the length of the array
+    }
+  })
+  return ans/len //return the average
 }
-
-module.exports = {average};
+module.exports = {average};//export the function
```

Comment:
Try a new test: The tests in `test/average.test.js` express the functionality of the `average` function.
The functionality accepts a sequence of readings from an IoT sensor. The sensor sends `NaN` when something goes wrong in measurement, which we have ignored.
Problem: If the sensor sends a lot of NaNs and very few readings, something is wrong. However, the `average` function ignores all the NaNs and still computes the average, as if nothing went wrong.
We need to add functionality to the average function to return `NaN`, when most of the input-numbers are `NaN`.
You need to:

- Reason out what 'most of the numbers' means: E.g., determine the number of NaNs in the input which would indicate that the sensor is unreliable. Mention any references you used to arrive at this logic
- Add a test that supplies such inputs and expects the output to be NaN
- Add code that satisfies this functionality, while not breaking existing functionality.

Reply:
For finding out if a sensor is faulty, we could determine it with the help of frequency of NaN values it gives. 
We can compare the few readings that we get with outliers as outliers are at an abnormal distance from other values(NaN in our case). Removing the outliers, will approximately leaves out with us 75% or more of NaN values. Hence, I have considered 75% as the checkpoint to evaluate the reliability of sensor.